web/src/app/schedules/ScheduleDetails.tsx

```diff
@@ -34,28 +34,60 @@ const query = gql`
   }
 `
 
-export const ScheduleCalendarContext = React.createContext({
-  onNewTempSched: () => {},
-  onEditTempSched: () => {},
-  onDeleteTempSched: () => {},
-  // ts files infer function signature, need parameter list
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  setOverrideDialog: (overrideVal) => {},
-})
+// todo - move this interface to more generic file when those are converted to typescript
+interface ScheduleCalendarOverrideDialogProps {
+  variantOptions: string[]
+  removeUserReadOnly: boolean
+  defaultValue: {
+    addUserID?: string
+    removeUserID?: string
+    start: string
+    end: string
+  }
+}
+
+interface ScheduleCalendarContextProps {
+  onNewTempSched: () => void
+  onEditTempSched: (v: TempSchedValue) => void
+  onDeleteTempSched: React.Dispatch<React.SetStateAction<null>>
+  setOverrideDialog: React.Dispatch<
+    React.SetStateAction<ScheduleCalendarOverrideDialogProps | null>
+  >
+}
+
+export const ScheduleCalendarContext =
+  React.createContext<ScheduleCalendarContextProps>({
+    onNewTempSched: () => {},
+    onEditTempSched: () => {},
+    onDeleteTempSched: () => {},
+    setOverrideDialog: () => {},
+  })
 
-export default function ScheduleDetails({ scheduleID }) {
+interface ScheduleDetailsProps {
+  scheduleID: string
+}
+
+export default function ScheduleDetails({
+  scheduleID,
+}: ScheduleDetailsProps): JSX.Element {
   const [showEdit, setShowEdit] = useState(false)
   const [showDelete, setShowDelete] = useState(false)
-  const [configTempSchedule, setConfigTempSchedule] = useState(null)
+  const [configTempSchedule, setConfigTempSchedule] = useState<
+    TempSchedValue | null | undefined
+  >()
```

Comment:
Yep. So this is one of the only functional changes I made. Before we were setting this to variable to type:
null = don't show tempSched window
true = show tempSched window, but with no passed in config (aka. it's a brand new temp schedule)
TempSchedValue = show tempSched, with a passed in config (aka. we're editing a current temp schedule)

So basically this variable is holding 2 pieces of information: 
A. Should we show the temp schedule window?, and 
B. What should we pass in as the current temp schedule?

And this was odd to convert to typescript because this useState would be of type <TempSchedValue | null | true>. Kind of a weird typing.

So instead it's now of type:
undefined = don't show tempSched window
null = show tempSched window, but with no passed in config (aka. it's a brand new temp schedule)
TempSchedValue = show tempSched, with a passed in config (aka. we're editing a current temp schedule)

It's still confusing though, I agree. Alternatively we could split this into 2 pieces of state: a `shouldShowTempSchedDialog` and an `activeTempSched`, or (and this would probably be the cleanest) we could make a generic custom hook that manages this kind of thing.

API could be like: { isOpen, onOpen, onClose } = useCustomDialog();

where the onOpen looks like (roughly):

```

function useCustomDialog<T>() {
  const [isOpen, setIsOpen] = useState();
  const [value, setValue] = useState<T>();

  const onOpen = (newValue) => {
    setValue(newValue);
    setIsOpen(true);
  }

  const onClose = () => {
    setValue(null);
    setIsOpen(false);
  }

  return {isOpen, value, onOpen, onClose}
}
```